core/API/Request.php

```diff
@@ -219,11 +220,10 @@ public function process()
             list($module, $method) = $this->extractModuleAndMethod($moduleMethod);
 
             list($module, $method) = self::getRenamedModuleAndAction($module, $method);
+            
+            PluginManager::getInstance()->checkIsPluginActivated($module);
 
-            if (!\Piwik\Plugin\Manager::getInstance()->isPluginActivated($module)) {
-                throw new PluginDeactivatedException($module);
-            }
-            $apiClassName = self::getClassNameAPI($module);
+            $apiClassName = $this->getClassNameAPI($module);
```

Comment:
Note: this reverts https://github.com/piwik/piwik/pull/7344/files but that's not wanted